static/app/components/replays/breadcrumbs/replayBreadcrumbOverview.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import range from 'lodash/range';
+import moment from 'moment';
+
+import {transformCrumbs} from 'sentry/components/events/interfaces/breadcrumbs/utils';
+import StackedContent from 'sentry/components/replays/stackedContent';
+import space from 'sentry/styles/space';
+import {Crumb, RawCrumb} from 'sentry/types/breadcrumbs';
+
+import {countColumns, formatTime, getCrumbsByColumn} from '../utils';
+
+const EVENT_STICK_MARKER_WIDTH = 2;
+
+interface Props {
+  data: {
+    values: Array<RawCrumb>;
+  };
+}
+
+type LineStyle = 'dotted' | 'solid' | 'none';
+
+function ReplayBreadcrumbOverview({data}: Props) {
+  const transformedCrumbs = transformCrumbs(data.values);
+
+  return (
+    <StackedContent>
+      {({width}) => (
+        <React.Fragment>
+          <Ticks
+            crumbs={transformedCrumbs}
+            width={width}
+            minWidth={20}
+            lineStyle="dotted"
+          />
+          <Ticks
+            crumbs={transformedCrumbs}
+            width={width}
+            showTimestamp
+            minWidth={50}
+            lineStyle="solid"
+          />
+          <Events crumbs={transformedCrumbs} width={width} />
+        </React.Fragment>
+      )}
+    </StackedContent>
+  );
+}
+
+function Ticks({
+  crumbs,
+  width,
+  minWidth = 50,
+  showTimestamp,
+  lineStyle = 'solid',
+}: {
+  crumbs: Crumb[];
+  lineStyle: LineStyle;
+  width: number;
+  minWidth?: number;
+  showTimestamp?: boolean;
+}) {
+  const startTime = crumbs[0]?.timestamp;
+  const endTime = crumbs[crumbs.length - 1]?.timestamp;
+
+  const startMilliSeconds = moment(startTime).valueOf();
+  const endMilliSeconds = moment(endTime).valueOf();
+  const duration = endMilliSeconds - startMilliSeconds;
+
+  const {timespan, cols, remaining} = countColumns(duration, width, minWidth);
+
+  return (
+    <TimelineMarkerList totalColumns={cols} remainder={remaining}>
+      {range(0, cols).map((_, i) => (
+        <TickMarker key={i} lineStyle={lineStyle}>
+          {showTimestamp && <small>{formatTime((i + 1) * timespan)}</small>}
+        </TickMarker>
+      ))}
+    </TimelineMarkerList>
+  );
+}
+
+const TimelineMarkerList = styled('ul')<{remainder: number; totalColumns: number}>`
+  /* Reset defaults for <ul> */
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  height: 100%;
+  width: 100%;
+
+  /* Layout of the lines */
+  display: grid;
+  grid-template-columns: repeat(${p => p.totalColumns}, 1fr) ${p => p.remainder}fr;
+  place-items: stretch;
+`;
+
+const TickMarker = styled('li')<{lineStyle: LineStyle}>`
+  border-right: 2px ${p => p.lineStyle} ${p => p.theme.gray100};
+  text-align: right;
+`;
+
+function Events({crumbs, width}: {crumbs: Crumb[]; width: number}) {
+  const totalColumns = Math.floor(width / EVENT_STICK_MARKER_WIDTH);
+  const eventsByCol = getCrumbsByColumn(crumbs, totalColumns);
+
+  return (
+    <div style={{paddingTop: space(4)}}>
```

Comment:
Can we make a styled component here instead of inline styles?

Reply:
yup, done